sites/public/pages/applications/preferences/all.tsx

```diff
@@ -0,0 +1,310 @@
+import React, { useCallback, useMemo, useState, useEffect } from "react"
+import { useForm } from "react-hook-form"
+import {
+  AlertBox,
+  Form,
+  FormCard,
+  ProgressNav,
+  Field,
+  t,
+  ExtraField,
+  PREFERENCES_FORM_PATH,
+  PREFERENCES_NONE_FORM_PATH,
+  ExpandableContent,
+  Button,
+  AppearanceStyleType,
+  resolveObject,
+  mapPreferencesToApi,
+  mapApiToPreferencesForm,
+} from "@bloom-housing/ui-components"
+import FormsLayout from "../../../layouts/forms"
+import FormBackLink from "../../../src/forms/applications/FormBackLink"
+import { useFormConductor } from "../../../lib/hooks"
+import { FormMetadataOptions } from "@bloom-housing/backend-core/types"
+
+const PreferencesAll = () => {
+  const { conductor, application, listing } = useFormConductor("preferencesAll")
+  const preferences = listing?.preferences
+
+  const currentPageSection = 4
+
+  // eslint-disable-next-line @typescript-eslint/unbound-method
+  const { register, setValue, watch, handleSubmit, errors, getValues, trigger } = useForm({
+    defaultValues: {
+      application: { preferences: mapApiToPreferencesForm(application.preferences) },
+    },
+  })
+
+  /*
+    It creates string path for each preference option - for error validation purposes
+  */
+  const preferenceOptionObjectPaths = useMemo(() => {
+    return preferences?.reduce((acc, item) => {
+      const preferenceName = item.formMetadata?.key
+      const optionPaths = item.formMetadata?.options?.map(
+        (option) => `${PREFERENCES_FORM_PATH}.${preferenceName}.${option.key}.claimed`
+      )
+
+      Object.assign(acc, {
+        [preferenceName]: optionPaths,
+      })
+
+      return acc
+    }, {})
+  }, [preferences])
+
+  const onSubmit = (data) => {
+    const body = mapPreferencesToApi(data)
+
+    conductor.currentStep.save(body)
+    conductor.routeToNextOrReturnUrl()
+  }
+
+  const buildOptionName = useCallback((metaKey: string, option: string) => {
+    return `${PREFERENCES_FORM_PATH}.${metaKey}.${option}.claimed`
+  }, [])
+
+  const allOptionFieldNames = useMemo(() => {
+    const keys = []
+    preferences?.forEach((preference) =>
+      preference?.formMetadata?.options.forEach((option) =>
+        keys.push(buildOptionName(preference?.formMetadata.key, option.key))
+      )
+    )
+
+    return keys
+  }, [preferences, buildOptionName])
+
+  const watchPreferences = watch(allOptionFieldNames)
+
+  function uncheckPreference(metaKey: string, options: FormMetadataOptions[]) {
+    const preferenceKeys = options?.map((option) => option.key)
+    preferenceKeys.forEach((k) => setValue(buildOptionName(metaKey, k), false))
+  }
+
+  /*
+    It creates values for household member select input (InputType.hhMemberSelect)
+  */
+  const hhMmembersOptions = useMemo(() => {
+    const { applicant } = application
+
+    const primaryApplicant = (() => {
```

Comment:
I used IIFE because we have conditions and some logic inside, so it prevents the use of `let` and I think it's more readable and it creates scope for `primaryApplicant`.